src/assets.rs

```diff
@@ -300,8 +295,10 @@ impl Assets {
             verified: false,
             temporary: false,
         };
-        let mut editor = AssetEditor::new(&mut self.store, asset.clone());
-        editor.update();
+        if let Ok(store_asset) = self.store.load(&asset.definition.code) {
+            let mut editor = AssetEditor::new(&mut self.store, asset.clone());
+            editor.update(store_asset);
```

Comment:
This doesn't do anything, since the editor is dropped without calling `save`, and anyways we overwrite the entry with `self.store.store(...)` on line 302. I think it should look more like
```rust
let asset = ...;
let store_asset = self.store.load(&asset.definition.code);
let mut editor = AssetEditor::new(&mut self.store, asset);
if let Ok(store_asset) = store_asset {
  editor.update(store_asset);
}
editor.save();
Ok(editor)
```